lurk-verify/src/lib.rs

```diff
@@ -0,0 +1,224 @@
+use lurk::{
+    field::LurkField,
+    proof::nova::PublicParams,
+    store::{ScalarPointer, ScalarPtr, Store},
+};
+use pasta_curves::pallas;
+use serde::{Deserialize, Serialize};
+use std::convert::TryFrom;
+use std::fs::File;
+use std::io::{self, BufReader, BufWriter};
+use std::path::Path;
+
+pub use crate::error::VerifyError;
+
+pub mod error;
+
+pub const DEFAULT_REDUCTION_COUNT: ReductionCount = ReductionCount::One;
+
+pub type S1 = pallas::Scalar;
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct VerificationResult {
+    pub verified: bool,
+}
+
+impl VerificationResult {
+    fn new(verified: bool) -> Self {
+        Self { verified }
+    }
+}
+
+// Number of circuit reductions per step, equivalent to `chunk_frame_count`
+#[derive(Clone, Copy, Hash, PartialEq, Eq, Serialize, Deserialize)]
+pub enum ReductionCount {
+    One,
+    Five,
+    Ten,
+}
+
+impl TryFrom<usize> for ReductionCount {
+    type Error = String;
+
+    fn try_from(count: usize) -> Result<Self, String> {
+        match count {
+            1 => Ok(ReductionCount::One),
+            5 => Ok(ReductionCount::Five),
+            10 => Ok(ReductionCount::Ten),
+            c => Err(format!("Unsupported Reduction Count: {c}")),
+        }
+    }
+}
+impl ReductionCount {
+    pub fn count(&self) -> usize {
+        match self {
+            Self::One => 1,
+            Self::Five => 5,
+            Self::Ten => 10,
+        }
+    }
+}
+
+#[derive(Serialize, Deserialize)]
+pub struct LurkProof<'a, F: LurkField> {
+    #[serde(bound(
+        serialize = "Claim<F>: Serialize",
+        deserialize = "Claim<F>: Deserialize<'de>"
+    ))]
+    pub claim: Option<Claim<F>>,
+    pub public_inputs: Vec<S1>,
+    pub public_outputs: Vec<S1>,
+    pub num_steps: usize,
+    #[serde(bound(
+        serialize = "lurk::proof::nova::Proof<'a>: Serialize",
+        deserialize = "lurk::proof::nova::Proof<'a>: Deserialize<'de>"
+    ))]
+    pub proof: lurk::proof::nova::Proof<'a>,
+    pub reduction_count: ReductionCount,
+}
+
+// Either a pair of input and expected output or a commitment opening, so the verifier can check that
+// they match the scalar public inputs and outputs
+#[derive(Clone, Debug, Serialize, Deserialize, PartialEq, Eq)]
+pub enum Claim<F: LurkField> {
+    Text(String, String),
```

Comment:
A general-purpose verifier needs to be able to deal with the env and cont I/O also. I guess it's fine if the `Claim` usage is less powerful, but considering that fcomm `Claim` is fully expressive, it seems a bit weird to flatten and reduce functionality here.

Reply:
More generally, what do we gain by not using the fcomm data types? It seems like those are pretty comprehensive. If we need something 'simpler' for a minimal verifier, maybe we can refactor them to remove whatever excess is strictly unnecessary. Then fcomm itself can use these types (perhaps moved into `lurk-rs` somewhere as a common source).